### PR TITLE
Updating Push and Store with API changes for AGDroid 3.0.0

### PR DESCRIPTION
**Pull request by secondsun**
Don't merge this until AGDroid 3.0.0 is in Central.

@matzew Do these updates make sense?  After we have 3.0.0 staged I will add the javadocs.

### CODE REVIEW

File: docs/guides/aerogear-android/push.asciidoc [hunk 1]
```diff
@@ -84,38 +84,57 @@ To enable the permissions (necessary for Android < 4.2 ie 95% of Android devices
...
     <intent-filter>
         <action android:name="com.google.android.c2dm.intent.RECEIVE" />
         <category android:name="[PUT YOUR APP PACKAGE HERE]" />
     </intent-filter>
 </receiver>
+<service
+    android:name="org.jboss.aerogear.android.unifiedpush.gcm.AeroGearGCMMessageReceiver"
+    android:permission="com.google.android.c2dm.permission.SEND">
+    <intent-filter>
+        <action android:name="com.google.android.c2dm.intent.RECEIVE" />
```
Comment (matzew) [inline docs/guides/aerogear-android/push.asciidoc]: they really still use `c2dm` ??? 🐑

File: docs/guides/aerogear-android/store.asciidoc [hunk 2]
```diff
@@ -30,16 +30,15 @@ Note: A Memory store will go away when the application is removed from memory an
 ----
 DataManager.config("mySQLStoreName", SQLStoreConfiguration.class)
            .withContext(getApplicationContext())
-           .forClass(MyModel.class)
-           .store();
+           .store(MyModel.class);
```
Comment (matzew) [inline docs/guides/aerogear-android/store.asciidoc]: ah! much nicer! 😄

File: docs/guides/aerogear-android/store.asciidoc [hunk 3]
```diff
@@ -30,16 +30,15 @@ Note: A Memory store will go away when the application is removed from memory an
...
 
 SQLStore store = (SQLStore) dataManager.get("mySQLStoreName"); 
 store.open(/*callback*/);
 // or
 store.openSync();
 ----
 
-Note: for an SQLStore you will need to make sure that 'open' has been called before you can access it.
+Note: for an SQLStore you should make sure that 'open' or 'openSync' have been called before accessing the store.  Otherwise, the store will be opened when a request is made which may impact performance.
```
Comment (matzew) [inline docs/guides/aerogear-android/store.asciidoc]: 👍

**Comment by matzew** [comment]
so far, it does look good.

we willmerge after the shizzle hits maven central :)

**Comment by matzew** [comment]
@secondsun we could merge to amster, which only updates staging

File: docs/guides/aerogear-android/push.asciidoc (hunk 1 above)
Comment (secondsun) [inline docs/guides/aerogear-android/push.asciidoc]: It is the permission name, yes.

**Comment by secondsun** [comment]
@matzew Ahhh, right.  Well I suppose I don't mind merging then.

File: docs/guides/aerogear-android/push.asciidoc (hunk 1 above)
Comment (matzew) [inline docs/guides/aerogear-android/push.asciidoc]: that's lame, especially their c2dm service is retired :)

**Comment by matzew** [comment]
sure, but let's wait until we at least have something staged :)

**Pull request merged**